modules/uid2IdSystem_shared.js

```diff
@@ -160,18 +464,137 @@ function refreshTokenAndStore(baseUrl, token, clientId, storageManager, _logInfo
       originalToken: token,
       latestToken: response.identity,
     };
+    let storedTokens = storageManager.getStoredValueWithFallback();
+    if (storedTokens?.originalIdentity) tokens.originalIdentity = storedTokens.originalIdentity;
     storageManager.storeValue(tokens);
     return tokens;
   });
 }
 
+function normalizeEmail(email) {
```

Comment:
Hi @dgirardi 
Yes, this is a translation of our server-side code. We decided to revisit this code and refactor it, make sure this will be covered by tests

> gmail special case. gmail.com is not the only domain used by google mail

It is a decision our UID2 team made, we will keep this as it is to make it consistent with server-side

thanks.

Reply:
Hi @dgirardi, hope you are well. We've refactored this function and added tests to cover this. Could you please take a review? Thanks